[Arnav Bajpai] iP

arnavbajpai opened the pull request:
# Krypto ChatBot ⚓

## CS2103 IP Project by **Arnav Bajpai**  
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://en.wikipedia.org/wiki/Dave_Allen_(comedian)))

Krypto is a command-line chatbot that helps you manage your tasks efficiently. It supports adding, deleting, marking, and searching for tasks.  

## Features  
- Add tasks: ToDos, Deadlines, and Events  
- Delete tasks  
- Mark tasks as done/undone  
- List all tasks  
- Search for tasks by keyword  
- Save and load tasks from a file  

## Krypto is
* text-based
* easy to learn
* ~~FAST~~ **SUPER FAST** to use

All you need to do is,
1. download it from [here](https://github.com/arnavbajpai/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😋

And it is **FREE**!

## Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
- [ ] Recording events and rescheduling

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) {
        new Krypto(FILE_PATH).run();
    }

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

arnavbajpai committed:
Add Krypto

arnavbajpai committed:
Add Echo

arnavbajpai committed:
Add List

arnavbajpai committed:
Add Mark

arnavbajpai committed:
Add Task

arnavbajpai committed:
Add Task Extensions

arnavbajpai committed:
Add Exception Handling

arnavbajpai committed:
Add Remove

arnavbajpai committed:
Add UI Testing

zedonggg commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,41 @@
+        return content.toString();
+    }
+
+    public void markTask() {
+        this.isDone = true;
+        this.sign = "X";
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println(this);
+    }
+    public void unmarkTask() {
+        this.isDone = false;
+        this.sign = "";
+        System.out.println("OK, I've marked this task as not done yet:");
+        System.out.println(this);
```
I like the use of OOP within your program. Do you think it would be better to return a String rather than print out? Your invalid and incomplete commands return Strings so perhaps it would be more organized to have everything return Strings to one method that can then separately print everything

zedonggg reviewed (commented):
LGTM. You're probably still in the process of doing the other iP tasks so all the best in getting those done! Some suggestions from me would be to split everything into packages, and further extract out all the methods into objects (such as different command types or Ui classes). And remember to add your javadoc comments! [👍]

matthanfoo commented on src/main/java/Krypto.java:
```diff
@@ -0,0 +1,107 @@
+            }
+            else if (first.equals("mark")) {
+                int id = Integer.parseInt(split[1]);
+                arr.get(id-1).markTask();
```
can consider adding white space before and after operators: operators should be surrounded by a space character.

matthanfoo commented on src/main/java/Krypto.java:
```diff
@@ -0,0 +1,107 @@
+            }
+            else if (first.equals("delete")) {
+                int id = Integer.parseInt(split[1]);
+                arr.remove(id -1);
```
slightly inconsistent spacing here as well

matthanfoo commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,41 @@
+        return content.toString();
+    }
+
+    public void markTask() {
```
agreed with @zedonggg, i think this breaks abstraction slightly as all prints should be executed by the Krypto bot rather than by the Task class itself

matthanfoo reviewed (commented):
really well organised code! just a few syntax change suggestions

arnavbajpai committed:
Implement file saving

arnavbajpai committed:
Merge Level-7 implementation

arnavbajpai committed:
Add Dates and Times

arnavbajpai committed:
Implement show command

arnavbajpai committed:
Merge Level-8 implementation

arnavbajpai committed:
Add more OOP

arnavbajpai committed:
Add packages

arnavbajpai committed:
Add gradle support

arnavbajpai committed:
Add gradle

arnavbajpai committed:
Implement JUnit testing

arnavbajpai committed:
Add jar file

arnavbajpai committed:
Add JavaDoc

arnavbajpai committed:
Merge branch 'branch-A-JavaDoc' into master

arnavbajpai committed:
Follow coding convention

arnavbajpai committed:
Merge branch 'branch-A-CodingStandard' into master

# Conflicts:
#	src/main/data/Krypto.txt
#	src/main/java/Krypto/Commands/AddCommand.java
#	src/main/java/Krypto/Commands/DeleteCommand.java
#	src/main/java/Krypto/Commands/MarkCommand.java
#	src/main/java/Krypto/Commands/ShowCommand.java
#	src/main/java/Krypto/Commands/UnmarkCommand.java
#	src/main/java/Krypto/Exceptions/IncompleteCommand.java
#	src/main/java/Krypto/IO/Storage.java
#	src/main/java/Krypto/IO/UI.java
#	src/main/java/Krypto/Task/Event.java
#	src/main/java/Krypto/Task/Task.java

arnavbajpai committed:
Add find command

arnavbajpai committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/Krypto/Task/Task.java
#	src/main/java/Krypto/Utils/TaskList.java

arnavbajpai committed:
Fix Level-9 bugs

arnavbajpai committed:
Update README.md

arnavbajpai committed:
Update README.md

arnavbajpai committed:
Add Krypto GUI basic

arnavbajpai committed:
Merge branch 'branch-Level-10'

arnavbajpai committed:
Do level-10 cleanup

arnavbajpai committed:
Add assertions

arnavbajpai committed:
Improve code quality

arnavbajpai committed:
Merge pull request #2 from arnavbajpai/branch-A-Assertions

arnavbajpai committed:
Merge branch 'master' into branch-A-CodeQuality

arnavbajpai committed:
Merge pull request #3 from arnavbajpai/branch-A-CodeQuality

Make code more readable

arnavbajpai committed:
Merge pull request #4 from arnavbajpai/branch-A-CodeQuality

Merge branch 'master' into branch-A-CodeQuality

arnavbajpai committed:
Merge branch 'branch-A-CodeQuality'

arnavbajpai committed:
Merge branch 'master' of https://github.com/arnavbajpai/ip

arnavbajpai committed:
Fix inconsistent exception handling with standardized messages

Exception handling lacks context and misses edge cases.

Better error messages help users understand what went wrong and help
developers debug issues faster. Standardized handling prevents bugs.

Let's,
* add operation context to exception messages
* handle missing command arguments consistently

arnavbajpai committed:
Add task rescheduling

arnavbajpai committed:
Add UI snapshot

arnavbajpai committed:
Update README.md

arnavbajpai committed:
Update README.md

arnavbajpai committed:
Update Ui

arnavbajpai committed:
Merge branch 'master' of https://github.com/arnavbajpai/ip

arnavbajpai committed:
Update README.md

arnavbajpai committed:
Final touches

arnavbajpai committed:
Merge branch 'master' of https://github.com/arnavbajpai/ip

arnavbajpai committed:
Update README.md

arnavbajpai commented:
# Krypto ChatBot ⚓

> "Your mind is for having ideas, not holding them." - David Allen ([source](https://en.wikipedia.org/wiki/Dave_Allen_(comedian)))

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ **SUPER FAST** to use

All you need to do is,

1. download it from [here](https://github.com/arnavbajpai/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😋

And it is **FREE**!

## Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
    public static void main(String[] args) {
        Application.launch(Krypto.class, args);
    }
}

arnavbajpai committed:
Update README.md

arnavbajpai committed:
Add more Junit testing

arnavbajpai committed:
Merge branch 'master' of https://github.com/arnavbajpai/ip
